cura/API/Machines.py

```diff
@@ -0,0 +1,139 @@
+#       api = CuraAPI()
+#       api.machines.addOutputDeviceToCurrentMachine()
+#       ```
+
+##  Since Cura doesn't have a machine class, we're going to make a fake one to make our lives a
+#   little bit easier.
+class Machine():
+    def __init__(self) -> None:
+        self.hostname = "" # type: str
+        self.group_id = "" # type: str
+        self.group_name = "" # type: str
+        self.um_network_key = "" # type: str
+        self.configuration = {} # type: Dict
+        self.connection_types = [] # type: List
+
+class Machines(QObject):
+
+    def __init__(self, application: "CuraApplication", parent = None) -> None:
+        super().__init__(parent)
+        self._application = application
+
+    @pyqtSlot(result="QVariantMap")
+    def getCurrentMachine(self) -> Machine:
+        fake_machine = Machine() # type: Machine
+        global_stack = self._application.getGlobalContainerStack()
+        if global_stack:
+            metadata = global_stack.getMetaData()
+            if "group_id" in metadata:
+                fake_machine.group_id = global_stack.getMetaDataEntry("group_id")
+            if "group_name" in metadata:
+                fake_machine.group_name = global_stack.getMetaDataEntry("group_name")
+            if "um_network_key" in metadata:
+                fake_machine.um_network_key = global_stack.getMetaDataEntry("um_network_key")
+
+            fake_machine.connection_types = global_stack.configuredConnectionTypes
+            
+        return fake_machine
+
+    ##  Set the current machine's friendy name.
+    #   This is the same as "group name" since we use "group" and "current machine" interchangeably.
+    #   TODO: Maybe make this "friendly name" to distinguish from "hostname"?
+    @pyqtSlot(str)
+    def setCurrentMachineGroupName(self, group_name: str):
+        Logger.log("d", "Attempting to set the group name of the active machine to %s", group_name)
+        global_stack = self._application.getGlobalContainerStack()
+        if global_stack:
+            # Update a GlobalStacks in the same group with the new group name.
+            group_id = global_stack.getMetaDataEntry("group_id")
+            machine_manager = self._application.getMachineManager()
+            for machine in machine_manager.getMachinesInGroup(group_id):
+                machine.setMetaDataEntry("group_name", group_name)
+
+            # Set the default value for "hidden", which is used when you have a group with multiple types of printers
+            global_stack.setMetaDataEntry("hidden", False)
+
+    ##  Set the current machine's configuration from an (optional) output device.
+    #   If no output device is given, the first one available on the machine will be used.
+    #   NOTE: Group and machine are used interchangeably.
+    #   NOTE: This doesn't seem to be used anywhere. Maybe delete?
+    @pyqtSlot(QObject)
+    def updateCurrentMachineConfiguration(self, output_device: Optional["PrinterOutputDevice"]) -> None:
```

Comment:
Is this really an `update`? Or is it semantically the same as making the given output device the active one?

Reply:
Honestly, I'm not sure. It looks like a weird hack where we force an update by emitting signals but it's not actually used anywhere as far as I can tell so it's tough to know what precisely the goal is.

I'd like to leave it in for now but will probably remove it in the near future.

Reply:
Do you want to add a TODO or maybe a warning here?